python/freetensor/core/remote_task_scheduler.py

```diff
@@ -0,0 +1,779 @@
+from __future__ import annotations
+import queue
+import threading
+import time
+import copy
+from typing import Any, List
+from typing import Dict
+'''
+task_type is specified
+1.search_task
+2.measure_task
+(minus means result)
+special param:
+0.other(transfer control)
+then check param
+    trans_c:(those are conbined with additional param: time_stamp)
+    0. logger
+    1. broadcast availability
+    2. return inavailability
+    3. new_task_request    
+'''
+'''
+How does this part work:
+1. commit a task via methods like RemoteTaskScheduler.remote_measure_submit()
+    in this part, the task will be transformed into a Task object, then:
+        1.1 a unique threading lock is acquired
+        1.2 the task is splitted and pushed into one of the submit queue(measure queue, search queue)
+        1.3 a message is broadcasted to all workers, telling them that new tasks are available to fetch
+
+2. a worker receive the broadcast in 1.3, and update the server's availability
+    if needed, the worker will send requests to all 
+    available servers for new tasks (currently, it 
+    means the number tasks in execution queue < 5)
+
+3. the client receive the request in 2, and send tasks to the worker(one for a time),
+    if no tasks are available, it will send
+    another message, telling that the server's 
+    submit queues are not available
```

Comment:
Since we are using RPC, do you mean *returning* the task (or the error message) to the worker? Why bother sending another request?

Reply:
OK, I see the following request send to the worker requests a return value.

Reply:
The resource may be occupied for a long time, so I return a value first and then wait for completion before returning the result

Reply:
Time is not a problem, as long as you make an asynchronous request. Actually this is called a *long poll*.